src/metrics.rs

```diff
@@ -0,0 +1,244 @@
+use anyhow::Error;
+use async_trait::async_trait;
+use chrono::{Duration, Utc};
+use fehler::throws;
+use octocrab::models::Repository;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use crate::google_sheets::Sheets;
+use crate::util;
+
+#[async_trait]
+pub trait Producer {
+    fn column_names(&self) -> Vec<String>;
+    fn spawn_producer_task(&self, tx: Sender<Vec<String>>);
+    async fn producer_task(&self, tx: Sender<Vec<String>>);
+}
+
+#[async_trait]
+pub trait Consumer {
+    async fn consume(&self, rx: &mut Receiver<Vec<String>>, column_names: Vec<String>);
+}
+
+pub struct ListReposForOrg {
+    org_name: String,
+}
+
+impl ListReposForOrg {
+    pub fn new(org_name: &str) -> Self {
+        ListReposForOrg {
+            org_name: String::from(org_name),
+        }
+    }
+}
+
+#[async_trait]
+impl Producer for ListReposForOrg {
+    fn column_names(&self) -> Vec<String> {
+        vec![String::from("Repository Name"), String::from("# of PRs")]
+    }
+
+    async fn producer_task(&self, tx: Sender<Vec<String>>) {
+        let octo = octocrab::instance();
+        let gh_org = octo.orgs(&self.org_name);
+
+        let repos: Vec<Repository> = match all_repos(&gh_org).await {
+            Ok(r) => r,
+            Err(e) => {
+                println!("Ran into an error while gathering repositories! {}", e);
+                vec![]
+            }
+        };
+
+        for repo in &repos {
+            match count_pull_requests(&self.org_name, &repo.name).await {
+                Ok(count_prs) => {
+                    if let Err(_) = tx
+                        .send(vec![String::from(&repo.name), count_prs.to_string()])
+                        .await
+                    {
+                        println!("receiver dropped!");
+                    }
+                }
+
+                Err(e) => {
+                    println!(
+                        "Ran into an issue while counting PRs for repository {}: {}",
+                        &repo.name, e
+                    );
+                }
+            }
+        }
+    }
+
+    // `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
+    fn spawn_producer_task(&self, tx: Sender<Vec<String>>) {
```

Comment:
this code does not compile due to the errors described in [this Zulip message](https://github-metrics.zulipchat.com/#narrow/stream/291433-general/topic/generalize.20producer.2Fconsumer/near/242951396)

Comments for error are included in code currently.